Sources/TuistKit/Commands/ScaffoldCommand.swift

```diff
@@ -0,0 +1,160 @@
+    var type: ErrorType { .abort }
+
+    case templateNotFound(String)
+    case templateNotProvided
+    case nonEmptyDirectory(AbsolutePath)
+
+    var description: String {
+        switch self {
+        case let .templateNotFound(template):
+            return "Could not find template \(template)"
+        case .templateNotProvided:
+            return "You must provide template name"
+        case let .nonEmptyDirectory(path):
+            return "Can't generate a template in the non-empty directory at path \(path.pathString)."
+        }
+    }
+}
+
+class ScaffoldCommand: NSObject, Command {
+    // MARK: - Attributes
+
+    static let command = "scaffold"
+    static let overview = "Generates new project based on template."
+    private let listArgument: OptionArgument<Bool>
+    private let pathArgument: OptionArgument<String>
+    private let templateArgument: PositionalArgument<String>
+    private var attributesArguments: [String: OptionArgument<String>] = [:]
+    private let subParser: ArgumentParser
+
+    private let templateLoader: TemplateLoading
+    private let templatesDirectoryLocator: TemplatesDirectoryLocating
+
+    // MARK: - Init
+
+    public required convenience init(parser: ArgumentParser) {
+        self.init(parser: parser,
+                  templateLoader: TemplateLoader(),
+                  templatesDirectoryLocator: TemplatesDirectoryLocator())
+    }
+
+    init(parser: ArgumentParser,
+         templateLoader: TemplateLoading,
+         templatesDirectoryLocator: TemplatesDirectoryLocating) {
+        subParser = parser.add(subparser: ScaffoldCommand.command, overview: ScaffoldCommand.overview)
+        listArgument = subParser.add(option: "--list",
+                                     shortName: "-l",
+                                     kind: Bool.self,
+                                     usage: "Lists available scaffold templates",
+                                     completion: nil)
+        templateArgument = subParser.add(positional: "template",
+                                         kind: String.self,
+                                         optional: true,
+                                         usage: "Name of template you want to use",
+                                         completion: nil)
+        pathArgument = subParser.add(option: "--path",
+                                     shortName: "-p",
+                                     kind: String.self,
+                                     usage: "The path to the folder where the template will be generated (Default: Current directory).",
+                                     completion: .filename)
+        self.templateLoader = templateLoader
+        self.templatesDirectoryLocator = templatesDirectoryLocator
+    }
+
+    func parse(with parser: ArgumentParser, arguments: [String]) throws -> ArgumentParser.Result {
+        guard arguments.count >= 2 else { throw ScaffoldCommandError.templateNotProvided }
```

Comment:
If we [make this one non-optional](https://github.com/tuist/tuist/pull/1092/files#diff-f6381ec395e0b3f45116e85579127124R58) you might not need to check whether the template is provided. That validation will be done automatically by the parser.